lib/ansible/modules/network/ios/ios_vlan.py

```diff
@@ -211,38 +211,49 @@ def map_params_to_obj(module):
     return obj
 
 
-def map_config_to_obj(module):
-    output = run_commands(module, ['show vlan'])
-    lines = output[0].strip().splitlines()[2:-1]
-
-    if not lines:
-        return list()
-
-    objs = list()
-
-    for l in lines:
-        splitted_line = l.strip().replace(",", "").split()
-        if splitted_line == []:
-            break
-        obj = {}
-        obj['vlan_id'] = splitted_line[0]
-        obj['name'] = splitted_line[1]
-        obj['state'] = splitted_line[2]
-
-        if obj['state'] == 'suspended':
-            obj['state'] = 'suspend'
-
-        obj['interfaces'] = []
-        if len(splitted_line) > 3:
-            interface = []
-            for i in range(3, len(splitted_line)):
-                interface.append(splitted_line[i].replace('Gi', 'GigabitEthernet'))
-            obj['interfaces'].extend(interface)
-        objs.append(obj)
+def parse_to_logical_rows(out):
+    started_yielding = False
+    cur_row = []
+    for l in out.splitlines()[2:]:
+        if not l:
+            """Skip empty lines."""
+            continue
+        if '0' < l[0] < '9':
+            """Line starting with a number."""
+            if started_yielding:
+                yield cur_row
+                cur_row = []  # Reset it to hold a next chunk
+            started_yielding = True
+        cur_row.append(l)
+
+    # Return the rest of it:
+    yield cur_row
+
+
+def map_ports_str_to_list(ports_str):
+    return list(filter(bool, (p.strip().replace('Gi', 'GigabitEthernet') for p in ports_str.split(', '))))
+
 
+def parse_to_obj(logical_rows):
+    first_row = logical_rows[0]
+    rest_rows = logical_rows[1:]
+    obj = re.match(r'(?P<vlan_id>\d+)\s+(?P<name>[^\s]+)\s+(?P<state>[^\s]+)\s+(?P<interfaces>.*)', first_row).groupdict()
+    obj['interfaces'] = map_ports_str_to_list(obj['interfaces'])
+    obj['interfaces'].extend(prts_r for prts in rest_rows for prts_r in map_ports_str_to_list(prts))
+    return obj
+
+
+def parse_vlan_brief(vlan_out):
+    objs = []
+    for logical_rows in parse_to_logical_rows(vlan_out):
```

Comment:
This name is misleading, because it's plural. But in fact `parse_to_logical_rows()` is a generator, which emits exactly one logical row per loop iteration. So if we were to keep this explicit loop it should've been renamed to `logical_row` or just `row` (because the generator name already contains a good explanation for what it emits).
But I'd go for just a list comprehension as stated in my comment one line below :)
It's kinda idiomatic Python way.